scripts/docker_build.sh

```diff
@@ -17,16 +17,54 @@
 # SPDX-License-Identifier: Apache-2.0
 # License-Filename: LICENSE
 
-DOCKER_ARGS=$@
+set -e -o pipefail
+
+DOCKER_ARGS=$*
 
 GIT_ROOT=$(git rev-parse --show-toplevel)
 GIT_REVISION=$(git describe --abbrev=10 --always --tags --dirty --match=[0-9]*)
+PLATFORM=${PLATFORM:-linux/amd64}
+ORG_BASE_NAME="${ORG_BASE_NAME:-oss-review-toolkit}"
+
+# Parse and transform versions from unique .versions file
+# shellcheck disable=SC1091
+source .versions
+VERSION_ARGS=$(xargs printf -- '--build-arg %s\n' < .versions)
+
+# Auxiliary function to build intermediary and final image
+# Usage:
+# docker_build <message> <target> <tag> <image_version>
+docker_build() {
+    echo "$1"
+    # shellcheck disable=SC2086
+    docker buildx build \
+        -f "$GIT_ROOT/Dockerfile" \
+        --target $2 \
+        --platform "$PLATFORM" \
+        --tag "$ORG_BASE_NAME/$3:latest" \
+        --tag "$ORG_BASE_NAME/$3:$4" \
+        --tag "ghcr.io/$ORG_BASE_NAME/$3:latest" \
+        --build-arg ORT_VERSION="$GIT_REVISION" \
+        $VERSION_ARGS \
+        $DOCKER_ARGS \
+        "$GIT_ROOT"
+}
+
+# Set NOBASE on terminal to skipt straight to ort container build
+# if all intermediary and base images are there
+if [ -z "${NOBASE+x}" ]; then
+    docker_build "Building base image" base base "$GIT_REVISION"
+    for lang in python ruby nodejs rust go haskell_stack android_cmd dart sbt; do
```

Comment:
Perhaps define this language list at the top in a global variable to make it stand out more if we need to add more options?